packages/cardano-graphql/src/operations/stakePools.graphql

```diff
@@ -85,6 +43,77 @@ fragment allStakePoolFields on StakePool {
       }
     }
   }
+  owners {
+    address
+  }
+  cost
+  margin {
+    numerator
+    denominator
+  }
+  vrf
+  relays {
+    __typename
+    ... on RelayByName {
+      hostname
+      port
+    }
+    ... on RelayByAddress {
+      ipv4
+      ipv6
+      port
+    }
+    ... on RelayByNameMultihost {
+      dnsName
+    }
+  }
+  poolRegistrationCertificate {
+    transaction {
+      hash
+    }
+  }
+  rewardAccount {
+    address
+  }
+  pledge
+  metadataJson {
+    hash
+    url
+  }
+}
+
+fragment allStakePoolFields on StakePool {
+  id
+  hexId
+  status
+  # Review: not sure this is correct. Consider this situation:
+  # pool registration tx at blockNo:1, specifying epoch:4
+  # pool registration tx at blockNo:2, specifying epoch:3
+  # Is stake pool active at epoch 3 and then switches to new parameters at epoch 4?
+  # Current implementation will ignore the 2nd transaction, as it's ordered by epoch first.
+  # I assume we also need to order by tx index in block as there can be multiple txes for registration of the same stake pool?
```

Comment:
Yes, that will ensure the latest update is returned, and if you drop the ordering by `sinceEpochNo` it will remove the issue you raised above. The query should probably be filtering to only upcoming epochs, not limited to 1 result, allowing the client to see the schedule of changes and make a determination on what to show the user

Reply:
I think querying past pool parameters is also needed. Users might want to evaluate if pools tend to change their fees etc.